tests/foreman/api/test_role.py

```diff
@@ -622,12 +818,105 @@ def test_positive_user_group_users_access_as_org_admin(self):
             3. Create two users without assigning roles while creating them
             4. Assign Organization A and Location A to both users
             5. Create an user group with above two users
+            6. Assign Org Admin role to User Group
 
         :expectedresults: Both the user should have access to the resources of
             organization A and Location A
 
         :CaseLevel: System
         """
+        org_admin = self.create_org_admin_role(
+            orgs=[self.role_org.id], locs=[self.role_loc.id])
+        userone_login = gen_string('alpha')
+        userone_pass = gen_string('alphanumeric')
+        user_one = entities.User(
+            login=userone_login,
+            password=userone_pass,
+            organization=[self.role_org.id],
+            location=[self.role_loc.id]
+        ).create()
+        self.assertEqual(userone_login, user_one.login)
+        usertwo_login = gen_string('alpha')
+        usertwo_pass = gen_string('alphanumeric')
+        user_two = entities.User(
+            login=usertwo_login,
+            password=usertwo_pass,
+            organization=[self.role_org.id],
+            location=[self.role_loc.id]
+        ).create()
+        self.assertEqual(usertwo_login, user_two.login)
+        ug_name = gen_string('alpha')
+        user_group = entities.UserGroup(
+            name=ug_name,
+            role=[org_admin.id],
+            user=[user_one.id, user_two.id]
+        ).create()
+        self.assertEqual(user_group.name, ug_name)
+        # Creating Subnets and Domains to verify if user really can access them
+        subnet = entities.Subnet(
+            name=gen_string('alpha'),
+            organization=[self.role_org.id],
+            location=[self.role_loc.id],
+        ).create()
+        domain = entities.Domain(
+            name=gen_string('alpha'),
+            organization=[self.role_org.id],
+            location=[self.role_loc.id]
+        ).create()
+        sc = ServerConfig(
+            auth=(userone_login, userone_pass),
+            url=ServerConfig.get().url,
+            verify=False
+        )
+        # User 1 Access Tests
+        with self.assertNotRaises(HTTPError):
+            entities.Domain(sc).search(
+                query={
+                    'organization-id': self.role_org.id,
+                    'location-id': self.role_loc.id
+                }
+            )
+            entities.Subnet(sc).search(
+                query={
+                    'organization-id': self.role_org.id,
+                    'location-id': self.role_loc.id
+                }
+            )
+        self.assertIn(
+            domain.id,
+            [dom.id for dom in entities.Domain(sc).search()]
```

Comment:
1. Why do you perform search for the second time here? You could just store results from lines above and ensure there's desired entity in output. 
2. The steps and check below are completely the same, you can use `for` loop to avoid duplication. Smth like that:
```python
for login, password in (
        (userone_login, userone_password),
        (usertwo_login, usertwo_password)):
    with self.subTest(login):
        sc = ServerConfig(
            auth=(login, password),
            url=ServerConfig.get().url,
            verify=False
        )
        [...]
```